mbed/source/main.cpp

```diff
@@ -0,0 +1,423 @@
+                    return;
+                }
+
+                error = _ble.gap().setAdvertisingParameters(_adv_handle, adv_parameters);
+                if (error) {
+                    print_error(error, "Gap::setAdvertisingParameters() failed");
+                    return;
+                }
+
+                _have_adv_handle = true;
+            }
+        } else {
+            // Use legacy advertising handle.
+            _adv_handle = ble::LEGACY_ADVERTISING_HANDLE;
+        }
+
+        // Set advertising payload.
+        _adv_data_builder.setFlags();
+        _adv_data_builder.setName(DEVICE_NAME);
+        error = _ble.gap().setAdvertisingPayload(_adv_handle, _adv_data_builder.getAdvertisingData());
+        if (error) {
+            print_error(error, "Gap::setAdvertisingPayload() failed");
+            return;
+        }
+
+        // Start advertising.
+        error = _ble.gap().startAdvertising(_adv_handle, _advertise_time);
+        if (error) {
+            print_error(error, "Gap::startAdvertising() failed");
+            return;
+        }
+    }
+
+    /** Set up and start scanning */
+    void scan()
+    {
+        update_state(bt_test_state_t::SCAN);
+
+        ble::ScanParameters scan_params;
+        scan_params.setOwnAddressType(ble::own_address_type_t::RANDOM);
+
+        ble_error_t error = _ble.gap().setScanParameters(scan_params);
+        if (error) {
+            print_error(error, "Gap::setScanParameters failed");
+            return;
+        }
+
+        error = _ble.gap().startScan(_scan_time);
+        if (error) {
+            print_error(error, "Gap::startScan failed");
+            return;
+        }
+
+        printf("Scanning for %ums\r\n", _scan_time.valueInMs());
+    }
+
+    void connect_peripheral()
+    {
+        update_state(bt_test_state_t::CONNECT_PERIPHERAL);
+        printf("Connected as peripheral\r\n");
+    }
+
+    void connect_master()
+    {
+        update_state(bt_test_state_t::CONNECT_MASTER);
+        printf("Connected as master\r\n");
+    }
+
+private:
+    /* Gap::EventHandler */
+
+    void onAdvertisingStart(const ble::AdvertisingStartEvent &event) override
+    {
+        if (_is_periodic) {
+            ble_error_t error = _ble.gap().setPeriodicAdvertisingParameters(
+                _adv_handle,
+                ble::periodic_interval_t(100),
+                ble::periodic_interval_t(1000)
+            );
+            if (error) {
+                print_error(error, "Gap::setPeriodicAdvertisingParameters() failed");
+                return;
+            }
+
+            error = _ble.gap().startPeriodicAdvertising(_adv_handle);
+            if (error) {
+                print_error(error, "Gap::startPeriodicAdvertising() failed");
+                return;
+            }
+
+            printf("Periodic advertising for %ums\r\n", _advertise_time.valueInMs());
+        } else {
+            printf("Advertising started for %ums\r\n", _advertise_time.valueInMs());
```

Comment:
```suggestion
            printf("Advertising started for %lums\r\n", _advertise_time.valueInMs());
```
or `PRIu64` perhaps?